torchvision/datasets/folder.py

```diff
@@ -51,21 +86,34 @@ def make_dataset(
             is_valid_file should not be passed. Defaults to None.
 
     Raises:
+        ValueError: In case ``class_to_idx`` is empty.
         ValueError: In case ``extensions`` and ``is_valid_file`` are None or both are not None.
+        FileNotFoundError: In case no valid file was found for any class.
 
     Returns:
         List[Tuple[str, int]]: samples of a form (path_to_sample, class)
     """
-    instances = []
     directory = os.path.expanduser(directory)
+
+    if class_to_idx is None:
+        _, class_to_idx = find_classes(directory)
+    elif not class_to_idx:
+        raise ValueError("'class_to_index' must have at least one entry to collect any samples.")
```

Comment:
This is BC breaking, but in a "right" way. Before `make_dataset` silently returned an empty list in case `class_to_idx` was empty. IMO it is a reasonable assumption that no user calls `make_dataset` without having at least a single class. If you disagree with this assumption, we can get BC back by `return []` here.

Reply:
I think this makes sense, especially for the folder dataset.
Is there any other dataset that inherits this other than videodatasets?

Reply:
Sounds good to me then

Reply:
To better explain myself: imagine a scenario where someone has a `class MyCoolNewDataset(DatasetFolder)` and they override `MyCoolNewDataset.find_classes` with a custom `class_to_idx` logic.

If they call `MyCoolNewDataset.make_dataset` while passing `None` to the `class_to_idx` parameter, what will be used is the `class_to_idx` logic from the `find_classes` **function**, instead of using the logic from the `find_classes` **method** - which is different since they overrode it.

Does that make sense?

To avoid such issues I'm suggesting to force the user to pass `class_to_idx` in `DatasetFolder.make_dataset`, or more accurately to raise an error if `None` is passed in `DatasetFolder.make_dataset`